ai/providers/openai.py

```diff
@@ -33,13 +32,22 @@ def get_models(self) -> dict:
     def generate_response(self, prompt: str, system_content: str) -> str:
         try:
             self.client = openai.OpenAI(api_key=self.api_key)
-            response = self.client.chat.completions.create(
+            response = self.client.responses.create(
                 model=self.current_model,
-                n=1,
-                messages=[{"role": "system", "content": system_content}, {"role": "user", "content": prompt}],
+                input=[
+                    {"role": "developer", "content": system_content},
+                    {"role": "user", "content": prompt},
+                ],
                 max_tokens=self.MODELS[self.current_model]["max_tokens"],
```

Comment:
```suggestion
                max_output_tokens=self.MODELS[self.current_model]["max_tokens"],
```

To match requirements of the `o4` model this change might be needed.

> Token limits may need to be bumped down...

I agree with this to avoid accident ramblings 💸  Your call on changing `max_tokens` to match this though!